compat/unistd.cc

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+
+#include "compat/socket.h"
+#include "compat/unistd.h"
+#include "compat/wserrno.h"
+
+// 2025 MinGW and pre-2022 MSVC do not define these
+#ifndef _S_IREAD
+#define _S_IREAD 0x0100
+#endif
+
+#ifndef _S_IWRITE
+#define _S_IWRITE 0x0080
+#endif
+
+int
+xclose(int fd)
+{
+    auto sock = _get_osfhandle(fd);
+    if (sock == intptr_t(INVALID_HANDLE_VALUE)) {
+        // errno is already set by _get_osfhandle()
+        return -1;
+    }
+
+    char l_so_type[sizeof(int)];
+    int l_so_type_siz = sizeof(l_so_type);
+    if (::xgetsockopt(sock, SOL_SOCKET, SO_TYPE, l_so_type, &l_so_type_siz) == 0) {
+        const auto result = closesocket(sock);
+        if (result == SOCKET_ERROR)
+            SetErrnoFromWsaError();
+        return result;
+    } else {
+        const auto result = _close(fd);
+        if (result)
+            SetErrnoFromWsaError();
+        return result;
+    }
+}
+
+int
+xgethostname(char *name, size_t namelen)
+{
+    auto result = ::gethostname(name, namelen);
```

Comment:
Please use `const` whenever possible, just like AAA.

```suggestion
    const auto result = ::gethostname(name, namelen);
```

Reply:
are you sure? That's an int under the hood

Reply:
Sorry, I do not understand how knowing the exact type is related to a missing `const`. These `const` "decorations" are usually not there for the compiler; modern compilers usually already know that a local variable is not modified and will optimize accordingly. These decorations are mostly there for _humans_ to ease code interpretation and reduce bugs related to unexpected/hidden variable modifications. In better languages, `const` is the default!

Except for function declarations, the corresponding rule of thumb is very simple -- if `const` compiles, use it. The idea and its justification is very similar (and even simpler) to AAA.

As I have mentioned before, those who develop a habit of typing `const auto` _first_, miss fewer cases.